src/columnsTypes/typeInheritance.js

```diff
@@ -1,28 +1,55 @@
 import {columnTypesMap} from './types.config';
 import {TableMap} from '../tablemap';
 
-export const typeInheritance = (view, node, pos) => {
-  if (!view || !node) return;
+/**
+ * Updates cells within table to inherit types from their column headers,
+ * setting attrs.type and rendering the type-appropriate content for cell.
+ *
+ * Call this whenever cells may have the wrong type, either because they are
+ * new or their column type was changed.
+ *
+ * @param tr A transaction to add the changes to
+ * @param tableStart The table's tableStart position, i.e., the pos of the first row
+ * (1 + pos of table itself) relative to tr.doc
+ * @return tr
+ */
+export const typeInheritance = (tr, tableStart) => {
+  // Subtract one to go from tableStart (start of first row) to actual table node's pos.
+  const table = tr.doc.nodeAt(tableStart - 1);
+
+  if (!table.attrs.headers || !table.maybeChild(0)) return tr;
+
+  // Store col types in colTypes (null if we should skip checking the column).
   // we don't allow cells merging, so we fill comfortable to check only the first row
-  const tableMap = TableMap.get(node);
-  const {tr} = view.state;
+  const tableMap = TableMap.get(table); // Will error if table is not a table
+  const colTypes = [];
   for (let col = tableMap.width - 1; col >= 0; col--) {
-    if (!node.child(0) || !node.child(0).maybeChild(col)) continue;
-    const header = node.child(0).child(col);
-    const colType = header.attrs.type;
+    if (!table.child(0).maybeChild(col)) colTypes[col] = null;
+    else {
+      const header = table.child(0).child(col);
+      colTypes[col] = header.attrs.type;
+    }
+  }
+
+  // Loop through cells in reverse node order, so that we don't have to map
+  // positions through previous steps or refresh tableMap.
+  // Since the order is row-major, this means reverse row order, then reverse column order.
+  for (let row = tableMap.height - 1; row >= 0; row--) {
+    if (!table.maybeChild(row)) continue;
+    for (let col = tableMap.width - 1; col >= 0; col--) {
+      const colType = colTypes[col];
+      if (colType === null || !table.child(row).maybeChild(col)) continue;
 
-    for (let row = tableMap.height - 1; row >= 0; row--) {
-      if (!node.child(row) || !node.child(row).maybeChild(col)) continue;
```

Comment:
The `!node.child(row)` check should be captured by line 38 above. I changed it to `maybeChild` because according to the docs, `child(row)` will throw an error if it's not a child, while `maybeChild` just returns null.

Reply:
oh ok I see sounds good